colcon_in_container/providers/multipass.py

```diff
@@ -0,0 +1,171 @@
+# Copyright (C) 2024 Canonical, Ltd.
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+from platform import processor
+import shutil
+import subprocess
+from typing import List
+
+from colcon_in_container.logging import logger
+from colcon_in_container.providers import exceptions
+from colcon_in_container.providers.provider import Provider
+import jinja2
+
+
+def _get_multipass_path():
+    return shutil.which('multipass')
+
+
+class MultipassClient(Provider):
+    """Multipass client interacting with the Multipass socket."""
+
+    def __init__(self, ros_distro):  # noqa: D107
+        super().__init__(ros_distro)
+
+        self.multipass_path = _get_multipass_path()
+        if not self.multipass_path:
+            raise exceptions.ProviderNotInstalledOnHostError(
+                'Multipass is not installed.'
+                'Please run `sudo snap install multipass`')
+
+        self._render_jinja_template()
+
+        if self._run(['info', self.instance_name]).returncode == 0:
+            self._clean_instance()
+
+        cpus = os.getenv('COLCON_IN_CONTAINER_MULTIPASS_CPUS', default='2')
+        mem = os.getenv('COLCON_IN_CONTAINER_MULTIPASS_MEMORY', default='2G')
+        disk = os.getenv('COLCON_IN_CONTAINER_MULTIPASS_DISK', default='256G')
```

Comment:
This on the other hand is way to much :D ! How about defaulting to e.g. 16G ?

Reply:
I took the default values from craft provider: https://github.com/canonical/craft-providers/blob/b0ada6a7410575550a16ee03521eabf1ec2d7751/craft_providers/multipass/multipass_instance.py#L337

Multipass won't take all that space on the disk if not necessary.